homeassistant/components/anthemav/config_flow.py

```diff
@@ -0,0 +1,109 @@
+"""Config flow for Anthem A/V Receivers integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+import anthemav
+from anthemav.connection import Connection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import (
+    CONF_MODEL,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEVICE_TIMEOUT_SECONDS,
+    DOMAIN,
+    EMPTY_MAC,
+    UNKNOWN_MODEL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+
+async def connect_device(user_input: dict[str, Any]) -> Connection:
+    """Connect to the AVR device."""
+    avr = await anthemav.Connection.create(
+        host=user_input[CONF_HOST], port=user_input[CONF_PORT], auto_reconnect=False
+    )
+    await avr.reconnect()
+    await avr.protocol.wait_for_device_initialised(DEVICE_TIMEOUT_SECONDS)
+    return avr
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Anthem A/V Receivers."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        if CONF_NAME not in user_input:
+            user_input[CONF_NAME] = DEFAULT_NAME
+
+        errors = {}
+
+        avr: Connection | None = None
+        try:
+            avr = await connect_device(user_input)
+        except OSError:
+            _LOGGER.error(
+                "Can't established connection to %s:%s",
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+            )
+            errors["base"] = "cannot_connect"
+        except asyncio.TimeoutError:
+            errors["base"] = "cannot_receive_deviceinfo"
+        except Exception as err:  # pylint: disable=broad-except
```

Comment:
Don't we know which errors that can come and be very specific on the possible errors?

Reply:
As far as I know, we don't actually expect any other exception than the 2 above, so do you think I can completely remove this broad except? How would the UI react if we don't catch an unexpected exception? 
The underlying library is calling create_connection from the current event_loop. Do you know how I could get an exhaustive list of exceptions returned by create_connection (sorry, my knowledge of python isn't as good as yours, so a bit struggling to find all the information). I couldn't find in the python documentation how to get the list of all the expected exception returned by this method. But all the test I've done, it is always returning OSError when connection can't be made. And timeout error is when we didn't received the expected command from the amplifier in a reasonable time. I'm happpy to completely drop the broad except